### DISCUSSION TITLE
STT not working properly on IOS

### DISCUSSION THREAD

**Discussion by Grzetan** [deepgram/community#746] [General help] [answered]
Hello, I have a problem with Speech to text (streaming) on IOS devices.
When I talk only the first 3/4 words get processed, after that no other responses are received from deepgram.
The same code which sends data and gathers it from microphone works fine on Android, Firefox and Chrome.
Does deepgram expect some kind of modified version of MP4 to be send in streaming API?

audio/ogg; codecs=opus - firefox

audio/webm;codecs=opus - chrome/android

audio/mp4 - ios

So I guess there is a problem somewhere with mp4 format?

Thanks in advance

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by SandraRodgers** [comment, marked as answer]
Are you sure that you are sending audio from the microphone continuously? If you send a 0 byte data, Deepgram will disconnect. You should do something like this to make sure you never send an empty data object:

```js
microphone.ondataavailable = (e) => {
      if (e.data.size > 0) {
        socket.send(e.data);
      }
```
